### DISCUSSION TITLE
Possibility to use Wifi hotspot for internet connection?

### DISCUSSION THREAD

**Discussion by RetoHB** [tesla-android/issue-tracker#221] [Ideas]
Build 2023.7.1 is just awesome! A big step forward! The only downside remaining for me is that every time I get in the car I have to plug in the phone for USB internet thetering.
Is there a possibility to implement some sort of wireless internet connection? I know bandwith is limited. May be we can use a different Wifi band? Or add a second Wifi adapter?

**Comment by mikegapinski** [comment]
Connecting to a Wi-Fi network while broadcasting another one for the car to connect to is not possible at all.

The only way to get rid of the tether is adding an external lte modem to the mix

**Reply by RetoHB** [reply]
But on my Samsung phone I can be connected to a Wifi network and at the same time share the connection as a hotspot. Is this a completely different thing?

**Reply by mikegapinski** [reply]
The Wi-Fi adapter in the pi does not support it

**Reply by RetoHB** [reply]
What if we add an USB adapter. This would also bring a better antenna.
An LTE modem requires an additional SIM card what I do not have...

**Reply by mikegapinski** [reply]
There are no drivers for them in Android. I am going for the CM4 soon, it has an external antenna connector.

The Wi-Fi chip in the Pi is decent. Most external ones under 50USD perform worse actually

**Reply by UziKhana** [reply]
does Android have drivers for ethernet? I have a mobile hotspot with ethernet port which could supply internet to rpi

**Reply by mikegapinski** [reply]
> does Android have drivers for ethernet? I have a mobile hotspot with ethernet port which could supply internet to rpi

Yes, it will work

**Reply by UziKhana** [reply]
thanks, i asked the question and then realized thats how i got the internet to rpi on the bench! silly me

**Reply by mikegapinski** [reply]
> thanks, i asked the question and then realized thats how i got the internet to rpi on the bench! silly me

You can also connect to the network provided by your router. Just set the DHCP range to something like I did. It should be  better than the Wi-Fi antenna in the pi

**Comment by ks0002544** [comment]
What about bluetooth connection? Android smartphone supports possibility to connect into the mobile internet using bluetooth of phone.

**Reply by thetangstr** [reply]
+1 would love to have bluetooth internet tether with the new wireless dongle

**Comment by sispeo29** [comment]
Hello
Another option could be to connect the car and the pi to the hotspot that provides the connection. As they will be in the same network, it should not be an issue for the tesla to reach the pi and be able to connect to internet through the hotspot.
The hostpot could then be a mobile phone sharing its internet connection, a wifi modem

**Comment by mikegapinski** [comment]
> Hello
> 
> Another option could be to connect the car and the pi to the hotspot that provides the connection. As they will be in the same network, it should not be an issue for the tesla to reach the pi and be able to connect to internet through the hotspot.
> 
> The hostpot could then be a mobile phone sharing its internet connection, a wifi modem

It won't work like this sadly

**Reply by sispeo29** [reply]
I missed something
for me, wifi is just the transport. Are there any additional protocol than TCP/IP going on between the car and the PI ?
Usually, as long as you have two (or more) devices connected to the same wifi, they should be able to talk to each other. No need to be one broadcasted by tesla android (except if tesla android needs to be the gateway in order to intercept all the trafic that is going on from the car)
Do you have some schema explaining the global picture of how tesla android is working ?
I am curious to understand
Thank you very much for your work by the way :)
